### PR TITLE
TGP-1478 Change validation for NIPHLS number in the Router

### PR DESCRIPTION
**Pull request by pauloncast**

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala [hunk 1]
```diff
@@ -114,6 +114,8 @@ object ValidationRules {
 
   private val actorIdPattern: Regex = raw"[A-Z]{2}\d{12,15}".r
   private val comcodePattern: Regex = raw".{6}(.{2}(.{2})?)?".r
+  private val niphlsPattern: Regex  = """^\d{4}$|^\d{5}$|^\d{6}$|^[A-Za-z]\d{5}$|^[A-Za-z]\d{6}$""".r
+//  private val niphlsPattern: Regex  = raw"^\d{4}$|^\d{5}$|^\d{6}$|^[A-Za-z]\d{5}$|^[A-Za-z]\d{6}".r
```
Comment (AmarkumarJagtap) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala]: can we remove the commented code if its not going to be used ever?

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/MaintainProfileRequest.scala [hunk 2]
```diff
@@ -37,11 +37,11 @@ object MaintainProfileRequest {
...
       (JsPath \ "niphlNumber")
-        .readNullable(lengthBetween(8, 8)))(MaintainProfileRequest.apply _)
+        .readNullable(validNiphls))(MaintainProfileRequest.apply _)
 
   implicit lazy val writes: OWrites[MaintainProfileRequest] =
     ((JsPath \ "actorId").write[String] and
       (JsPath \ "ukimsNumber").write[String] and
       (JsPath \ "nirmsNumber").writeNullable[String] and
-      (JsPath \ "niphlsNumber").writeNullable[String])(unlift(MaintainProfileRequest.unapply))
+      (JsPath \ "niphlNumber").writeNullable[String])(unlift(MaintainProfileRequest.unapply))
```
Comment (AmarkumarJagtap) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/MaintainProfileRequest.scala:46]: can we keep the name consistent everywhere if its going to be `niphlNumber ` and not `niphlsNumber ` ?

**Review by AmarkumarJagtap — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/MaintainProfileRequest.scala (hunk 2 above)
Comment (pauloncast) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/MaintainProfileRequest.scala:46]: The correct one is `niphlNumber` as updated here. It is updated everywhere else or is there anyone missing?

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala (hunk 1 above)
Comment (pauloncast) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala]: updated

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala [hunk 3]
```diff
@@ -114,6 +114,7 @@ object ValidationRules {
 
   private val actorIdPattern: Regex = raw"[A-Z]{2}\d{12,15}".r
   private val comcodePattern: Regex = raw".{6}(.{2}(.{2})?)?".r
+  private val niphlsPattern: Regex  = """^\d{4}$|^\d{5}$|^\d{6}$|^[A-Za-z]\d{5}$|^[A-Za-z]\d{6}$""".r
```
Comment (william-eyre) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala:117]: You need to include 7 characters in this check as well

**Review by william-eyre — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala (hunk 3 above)
Comment (william-eyre) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala:117]: I also think you could simplify the regex with this: 
^([a-zA-Z0-9]){4,7}$

Comment (pauloncast) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala:117]: The current regex accepts 7 characters like this S123456. The suggested regex accepts S1234 - Alpha-numeric-5, and S123 - Alpha-numeric-4, which are not part of what the ticket specified, if those are acceptable, we need to update the ticket as well to describe all possible acceptable patterns

**Review by AmarkumarJagtap — APPROVED** [review]

**Review by william-eyre — APPROVED** [review]

**Pull request merged**